resolwe_api/resolwe.py

```diff
@@ -283,32 +307,36 @@ def upload(self, process_name, name='', descriptor=None, descriptor_schema=None,
 
         """
         # This is temporary solution: map process name to it's ID:
-        proc_name_to_id = dict([(x['name'], x['slug']) for x in self.processes()])
+        proc_name_to_slug = dict([(x['name'], x['slug']) for x in self.processes()])
 
         p = self.processes(process_name=process_name)
 
         if len(p) == 1:
             p = p[0]
+        elif len(p) > 1:
+            # Multiple processors with same slug, but different versions:
+            versions = ([int(x['version']) for x in p])
+            # Get index of the latest processor version:
+            top_index = sorted(enumerate(versions), key=lambda x:x[1])[-1][0]
+            p = p[top_index]
         else:
-            Exception('Invalid process name {}'.format(process_name))
-
-        for field_name, field_val in fields.items():
-            if field_name not in p['input_schema']:
-                Exception("Field {} not in process {} inputs".format(field_name, p['name']))
-
-            if find_field(p['input_schema'], field_name)['type'].startswith('basic:file:'):
-                if not os.path.isfile(field_val):
-                    Exception("File {} not found".format(field_val))
+            raise Exception('Invalid process name: {}'.format(process_name))
```

Comment:
The comment is about messages. Here you write `'Invalid process name: {}'`, but maybe you should write `'invalid process name: {}` to be consistent with the lowercase message in line 280.